packages/slice-machine/server/src/api/services/getEnv.ts

```diff
@@ -39,17 +33,18 @@ function validate(config: Models.Manifest): ConfigErrors {
   return errors;
 }
 
-function extractRepo(parsedRepo: ParseResult): string {
-  switch (parsedRepo.type) {
-    case ParseResultType.Listed:
-      if (parsedRepo.labels.length) {
-        return parsedRepo.labels[0];
-      }
-      if (parsedRepo.subDomains.length) {
-        return parsedRepo.subDomains[0];
-      }
-    default:
-      return "";
+function extractRepo(apiEndpoint: Manifest["apiEndpoint"]): string {
+  try {
+    const url = new URL(apiEndpoint);
+    const host = url.host; // myrepo.prismic.io
+    const repository = host.split(".")[0]; // myrepo
+    return repository;
+  } catch (e) {
+    // Should already be covered by the start script.
+    console.error(
+      "It seems your repository API endpoint in your SM.json is wrong. Please start machine again to get instructions on how to fix it."
+    );
+    process.exit(0);
```

Comment:
pretty sure we don't want the server to stop running if the urls not valid

Reply:
It's something only us would see, a user would have to have "wroom.test" in his API endpoint 😅.
i figured as devs, we might want to know as soon as possible if something is broken in our dev env.

What do you think should be the correct behaviour ?

Reply:
pretty sure etEnv is used in a few places, do we want it to exit if the apiEndpointIs wrong?